### PR TITLE
[Quantum]Add Azure quantum api key support

### PR DESCRIPTION
**Pull request by 04diiguyi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Quantum

File: src/quantum/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+1.0.1
```
Comment (shenyingjun) [inline src/quantum/HISTORY.rst:6]: this need to be bX since we are still in preview

File: src/quantum/azext_quantum/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # This is the version reported by the CLI to the service when submitting requests.
 # This should be in sync with the extension version in 'setup.py', unless we need to
 # submit using a different version.
-CLI_REPORTED_VERSION = "1.0.0b1"
+CLI_REPORTED_VERSION = "1.0.1"
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/__init__.py:16]: same

File: src/quantum/setup.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 # This version should match the latest entry in HISTORY.rst
 # Also, when updating this, please review the version used by the extension to
 # submit requests, which can be found at './azext_quantum/__init__.py'
-VERSION = '1.0.0b1'
+VERSION = '1.0.1'
```
Comment (shenyingjun) [inline src/quantum/setup.py:20]: same

File: src/quantum/azext_quantum/operations/workspace.py [hunk 4]
```diff
@@ -380,3 +385,62 @@ def clear(cmd):
...
+def enable_keys(cmd, resource_group_name=None, workspace_name=None, enable_key=None):
+    """
+    Update the default Azure Quantum workspace.
+    """
+    client = cf_workspaces(cmd.cli_ctx)
+    info = WorkspaceInfo(cmd, resource_group_name, workspace_name, None)
+    if (not info.resource_group) or (not info.name):
+        raise ResourceNotFoundError("Please run 'az quantum workspace set' first to select a default Quantum Workspace.")
+
+    if (enable_key != "True") and (enable_key != "true") and (enable_key != "False") and (enable_key != "false"):
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/operations/workspace.py:434]: nit: we can define accepted values as a list and use "in" operator. some thing like if (enable_key !in ["true", "True"])

File: src/quantum/azext_quantum/operations/workspace.py [hunk 5]
```diff
@@ -228,6 +229,10 @@ def create(cmd, resource_group_name, workspace_name, location, storage_account,
     # Until the "--skip-role-assignment" parameter is deprecated, use the old non-ARM code to create a workspace without doing a role assignment
     if skip_role_assignment:
         _add_quantum_providers(cmd, quantum_workspace, provider_sku_list, auto_accept, skip_autoadd)
+        properties = WorkspaceResourceProperties()
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/operations/workspace.py:232]: why this change only applies to this code path when skip_role_assignment = true?

File: src/quantum/azext_quantum/_help.py [hunk 6]
```diff
@@ -283,3 +283,39 @@
         text: |-
             az quantum workspace show -g MyResourceGroup -w MyWorkspace
 """
+
+helps['quantum workspace update'] = """
+    type: command
+    short-summary: Enable or disable Azure Quantum workspace api keys.
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/_help.py:289]: quantum workspace update should be broader than just enable/disable key.
Maybe something like
short-summary: Update the given (or current) Azure Quantum workspace.

File: src/quantum/azext_quantum/_help.py [hunk 7]
```diff
@@ -283,3 +283,39 @@
...
+        text: |-
+            az quantum workspace update --enable-api-key True
+      - name: Disable a provided Azure Quantum workspace api keys.
+        text: |-
+            az quantum workspace update --enable-api-key False
+"""
+
+helps['quantum workspace keys list'] = """
+    type: command
+    short-summary: Display Azure Quantum workspace api keys.
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/_help.py:301]: List api keys  for the given (or current) Azure Quantum workspace.

File: src/quantum/azext_quantum/_help.py [hunk 8]
```diff
@@ -283,3 +283,39 @@
...
+    examples:
+      - name: Show the currently selected default Azure Quantum workspace api keys.
+        text: |-
+            az quantum workspace keys list
+"""
+
+helps['quantum workspace keys regenerate'] = """
+    type: command
+    short-summary: Regenerate Azure Quantum workspace api key.
```
Comment (shenyingjun) [inline src/quantum/azext_quantum/_help.py:310]: Regenerate api key for the given (or current) Azure Quantum workspace.

File: src/quantum/azext_quantum/_help.py (hunk 7 above)
Comment (shenyingjun) [reply src/quantum/azext_quantum/_help.py:301]: Let's try to word it consistently with existing doc  https://learn.microsoft.com/en-us/cli/azure/quantum/workspace?view=azure-cli-latest

File: src/quantum/azext_quantum/operations/workspace.py (hunk 5 above)
Comment (04diiguyi) [reply src/quantum/azext_quantum/operations/workspace.py:232]: Because our workspace property has been changed since last version. The regular way to use arm client works since we update the client. However,  skip role assignment does not directly use arm client, it uses another endpoint which will break the since the workspace class structure has been changed

**Comment by 04diiguyi** [comment]
Can someone manually trigger the integration tests to rerun? The service is deployed.

**Review by xinyi-joffre — APPROVED** [review]

**Review by necusjz — APPROVED** [review]
